docs/source/federation-guide/federated-schema-design.mdx

```diff
@@ -0,0 +1,545 @@
+
+### Create Semantically Meaningful Interfaces
+
+A common misuse of interfaces is to use them simply to express a contract for shared fields between types. While this is certainly an aspect of their intended use, they should only be used when you need to return an object or a set of objects from a field *and* those objects may represent a variety of different types with some fields in common. For example:
+
+```graphql
+interface Pet {
+  breed: String
+}
+
+type Cat implements Pet {
+  breed: String
+  extraversionScore: Int
+}
+
+type Dog implements Pet {
+  breed: String
+  activityLevelScore: Int
+}
+
+type Query {
+  familyPets: [Pet]
+}
+```
+
+In this schema, the `familyPets` query returns a list of cats and dogs, with a guarantee that the `breed` field will be implemented on both the `Cat` and `Dog` types. A client can then query for these types' shared fields as usual, or use inline fragments for the `Cat` and `Dog` types to fetch their type-specific fields:
+
+```graphql
+query GetFamilyPets {
+  familyPets {
+    breed
+    ... on Cat {
+      extraversionScore
+    }
+    ... on Dog {
+      activityLevelScore
+    }
+  }
+}
+```
+
+If there was no use case for querying both cats and dogs simultaneously to return both types from a single operation, then the `Pet` interface wouldn't serve any notable purpose in this schema. Instead, it would add overhead to schema maintenance by requiring that the `Cat` and `Dog` types continue to adhere to this interface as they evolve, but with no functional reason as to why they should continue conforming to `Pet`.
+
+What's more, the overhead for maintaining both interface and union types is amplified when dealing with federated data graphs. Where interfaces and unions are shared as value types across schemas, they become cross-cutting concerns (which we'll address further in a later section). Further, interfaces may also be entities in a federated data graph, so challenging decisions may need to be made about which service ultimately "owns" interface entities and whether the services that implement them in a schema can adequately resolve all the types that belong to that interface.
+
+While interfaces are abstract types, they should ultimately represent something concrete about the relationship they codify in a schema and they should indicate some shared behavior among the types that implement them. Satisfying this baseline requirement can help guide your decisions about where to use interfaces selectively in your federated schemas.
+
```

Comment:
I definitely think we should add a note/reminder that using an interface will require that subgraph to have a `__resolveType` resolver per the GraphQL spec

>_*Note*: Using an interface in any subgraph will require that subgraph to have a `__resolveType` implementation as per the GraphQL spec. This means that subgraph must have knowledge on how to resolve that type._

@prasek might want to have some input here in case we want to say a little something about how we're thinking about this

Reply:
Added a note about this in the revised version.